prepare_dataset.py

```diff
@@ -186,6 +193,48 @@ def resize_and_crop(data_prep_local_dir, target_size, image_cropping_params):
                                 (Path(data_prep_local_dir, 'resized', scan, 'annotations', scan_annotation_files[
                                     image_ind].name).as_posix()).replace('.', ('_crop' + str(counter_crop) + '.')))
                             counter_crop += 1
+                elif image_cropping_params['type'] == 'class':  # 'smart' crop: do not train with pad, some overlap okay (still aug'd)
+                    assert 'num_pos_per_class' in image_cropping_params
+                    assert 'num_neg_per_class' in image_cropping_params
+                    assert image_cropping_params['num_pos_per_class'] > 0  # logical choice
+                    assert image_cropping_params['num_pos_per_class'] <= 36  # suits 4600 x 2048 img with 512 x 512 target
+                    assert image_cropping_params['num_neg_per_class'] >= 0  # logical choice if 0
+                    assert image_cropping_params['num_neg_per_class'] <= 36  # suits 4600 x 2048 img with 512 x 512 target
+                    class_pos_min_threshold = 1  # min thresh that defines class pos pixel qty per crop
```

Comment:
To help our future selves -- constants like this should be global variables

Reply:
actually moved this threshold to dataset config file because it should be captured in metadata